src/fides/lib/db/base_class.py

```diff
@@ -151,7 +151,9 @@ def filter(
         return db.query(cls).filter(conditions)
 
     @classmethod
-    def create(cls: Type[T], db: Session, *, data: dict[str, Any]) -> T:
+    def create(
+        cls: Type[T], db: Session, *, data: dict[str, Any], check_name: bool = True
```

Comment:
i didn't love having to change this core method, but i am hoping my changes here shouldn't be disruptive in any way, given the default value. automated testing seems to back that hope up.

i'd started on trying to work around this constraint but everything just felt very contrived. i do think adding the optionality here is ultimately an improvement, i don't think we need to be strict in requiring all of our classes/tables to have their `name` column in a unique constraint.

Reply:
I agree - this was more for those models like `Policy` which have a readable `key` created for it. Part of the uniqueness on the name was because if there was no key supplied in the request we could create it from the name.  And then there were more models that we wanted this same behavior on. 

But a lot of our new models don't have this additional human readable key added to it, so I like this addition.  It's now feeling like this `name`/`key` behavior should be organized into it's own mixin just used on certain models instead of it being the default behavior. Not today's concern though.